packages/grid/x-data-grid/src/hooks/features/sorting/useGridSorting.ts

```diff
@@ -204,6 +204,31 @@ export const useGridSorting = (
     [apiRef],
   );
 
+  const setRowIndex = React.useCallback<GridSortApi['setRowIndex']>(
+    (rowId, targetIndexPosition) => {
+      const allRows = gridSortedRowIdsSelector(apiRef);
```

Comment:
You are still updating the sorting list.
Was updating `state.rows.ids` with the same value as `updatedRows` if this function + cleaning to sorting model to complicated or is it incompatible with something ?

Reply:
I didn't understand the suggestion. From what I saw the order of the rows is taken from the `state.sorting.sorterRows`. the other `state.rows.ids` isn't used to determine the actual order, no? Maybe I'm missing something?

Reply:
My proposal was when you reorder the column:

1. Update `state.rows.ids` with the actual order you now have (the same value you are currently storing in `state.sorting.sortedRows`)
2. Remove all the sort model items

That way, the sort engine will just take the order of `state.rows.ids` because it has no sorting rule.

Advantages:

- If the user reorder a row, the add a sorting rule, then remove it, it will not loose the reorder he did
- We are not updating a derived state that could be changed (for instance if a user edit a cell, it will re-apply the sorting and thus loose the reorder)

Disadvantages:

- If the sort model is being controlled, it can be harder to manage

For me the subject is the same as #4160
If we remove the automatic sorting / filtering when the data is partially update
Then updating `state.sorting.sortedRows` can be viable (and for the 1st advantage, the user can manually re-inject its rows inside the `props.rows` if he wants when the order changes).
If not, the editing part will probably be problematic.

Reply:
Updated. I also moved this API to the `useGridRows`.